src/policy/protect_payload.rs

```diff
@@ -192,3 +229,31 @@ impl ForwardingRule {
         self
     }
 }
+
+// ———————————————————————————————— Hash primitive ———————————————————————————————— //
+
+fn hash_payload(size_to_hash: usize, pc_start: usize) -> [u8; 32] {
+    let payload_start: usize = 0x80400000;
+    let payload_end: usize = 0x80400000 + size_to_hash;
+
+    assert!(
+        payload_start <= payload_end,
+        "Invalid memory range for payload hashing"
+    );
+
+    unsafe {
+        let mut hasher = Sha3::v256();
+
+        let payload_content =
+            slice::from_raw_parts(payload_start as *const u8, payload_end - payload_start);
+        hasher.update(payload_content);
+
+        let pc_start = slice::from_raw_parts(pc_start as *const u8, 4);
+        hasher.update(pc_start);
```

Comment:
We want to hash the *value* of the PC, not the *data pointed by* the PC. If you hash the data pointed by there is very little guarantees, as there could be lots of entry points starting with the same instruction.

If you need an array of bytes for the hasher, you can use `pc_start.to_le_bytes()`.